### ISSUE TITLE
Query to find all object with a certain custom field value

### ISSUE THREAD

**Issue by floreasy** [owen2345/camaleon-cms#673]
Hello,

My request is, I have different Post Types and some of them are linked. ( I have my post type "place" and my post type "news", one news is linked to a place via a custom field ) 
On my "Place" page, I'd like to list all the news related to it. What I do for the moment is I created a Helper function, which finds all the news for the place. My worry is I'm doing a query which is too long to execute : 
`mesAnims = CamaleonCms::Site.first.posts.sort_by_field("date-de-debut", "desc").where(post_type: 69)`
In other words, i find all the news. And then I do a loop to get only the ones for my current item.
`lesAnims = []

      mesAnims.each do |anim|

        if !anim.get_fields('bistrot-associe',0)[0].nil? && anim.get_fields('bistrot-associe',0)[0] == @post.id.to_s

            lesAnims.push(anim)

        end

      end` 
I would like to know if I could avoid this loop to directly add a "Where mycustomfieldslug = myvalue" . 
Thanks a lot,

**Comment by owen2345** [comment]
Hi @floreasy I improved this part and also I added small docs (sorting, filtering, preloading) here:
http://camaleon.tuzitio.com/documentation/category/40756-uncategorized/custom-fields-1.html
Please review and let me know if it works well for you.

**Comment by floreasy** [comment]
Hi @owen2345 and thanks a lot for improving this documentation. 
However, I just do not get one thing, in this : 

`Cama::PostType.first.posts.includes(:custom_field_values).merge(CamaleonCms::CustomFieldsRelationship.where(custom_field_slug: 'my_slug_key'))`

What is "custom_field_slug:" ? Is it like the name of the slug ? For example if the slug is like a city, would it be something like : `where(my_slug_for_city: "Paris"` ? Or am I wrong ? 

Hopefully then I'll be OK with my queries :) 

Thanks a lot again,

Flo

**Comment by floreasy** [comment]
By the way, when i use `.includes(:custom_field_values)` I have an error : 

`The association scope 'custom_field_values' is instance dependent (the scope block takes an argument). Preloading instance dependent scopes is not supported. `

**Comment by owen2345** [comment]
custom_field_slug is the DB column name to save the custom_field key.
Example:
```
Cama::PostType.first.posts.includes(:custom_field_values).merge(CamaleonCms::CustomFieldsRelationship.where(custom_field_slug: 'my_slug_for_city', value: 'Paris'))
# Filtering all posts that contains custom field key = "my_slug_for_city" and custom field value is "Paris".
```
Check image for the structure:
<img width="1190" alt="screen shot 2017-08-01 at 9 24 45 am" src="https://user-images.githubusercontent.com/1148415/28827366-574ed674-769b-11e7-80a8-c8b826c58018.png">

Please bundle update, I released another version with the changes *.11

**Comment by floreasy** [comment]
Yes, I just found out 5 minutes ago and now I see it on your comment :) 

Anyway, thanks a lot, and indeed I had also to bundle update. 

Thank you !
